storage-proofs-core/src/compound_proof.rs

```diff
@@ -77,12 +77,15 @@ where
         }
     }
 
-    /// prove is equivalent to ProofScheme::prove.
-    fn prove<'b>(
+    // helper function it is almost the same as prove
+    // but with a taskType parameter which is used in bellperson to define
+    // the priority of the task and handling it accordingly.
+    fn prove_with_type<'b>(
         pub_params: &PublicParams<'a, S>,
         pub_in: &S::PublicInputs,
         priv_in: &S::PrivateInputs,
         groth_params: &'b groth16::MappedParameters<Bls12>,
+        post_type: Option<BellTaskType>,
```

Comment:
(unrelated to @porcuquine's comment) It looks like you always pass in a type, so I don't think it needs to be wrapped in a `Option`.

Reply:
> `CompoundProof` is an abstraction that doesn't (and shouldn't) know anything whatsoever about specific proof types. So mention of `post_type` here, `MerkleTree` later, etc. is by definition inappropriate. It might be that `post_type` is just misnamed (will look more closely) — but the point stands: if we are referencing specific proof types of any kind here then something is wrong.

@porcuquine  I did not like the idea of refactoring that trait neither, as it seems to be a super Trait from which other traits inherit. but, at the moment it looked like the cleaning way to modify and pass the information as a task_type(which at the moment encodes deadlines and possible exclusive resources) to the scheduler, without having to refactor many other places.

@vmx Initially, it was defined as an optional param, I set it to MerkeProof, but I am not sure if that type should encompass all the processes that use the GPU but that are not Winning/Window post. It is important because all those processes will end up using the GPU assigned to MerkeTree operations.